spec/unit/lib/cloud_controller/diego/docker/lifecycle_protocol_spec.rb

```diff
@@ -118,6 +118,34 @@ module Docker
                 end
               end
             end
+
+            context 'when root user not allowed' do
+              before do
+                TestConfig.override(allow_docker_root_user: false)
+              end
+
+              context 'and the process does not set a user' do
+                let(:app) { AppModel.make(:docker, { droplet: }) }
+                let(:process) { ProcessModel.make(:docker, { app: }) }
+
+                context 'and the droplet docker execution metadata sets the root user' do
+                  let(:droplet_execution_metadata) { '{"entrypoint":["/image-entrypoint.sh"],"user":"root"}' }
```

Comment:
We had discussed offline the case where the docker user is set to `0` (linux user id for the `root` user). Do we need to cover that case here also?